internal/controller/pipelinerollout_controller.go

```diff
@@ -178,6 +183,10 @@ func (r *PipelineRolloutReconciler) processPipelineRollout(ctx context.Context,
 	if requeue {
 		return ctrl.Result{Requeue: true, RequeueAfter: 30 * time.Second}, nil
 	}
+
+	// generate the metrics for the Pipeline.
+	r.customMetrics.IncPipelineMetrics(pipelineRollout.Name, pipelineRollout.Namespace)
```

Comment:
thinking about this one - it's possible we could decide to requeue when everything has gone well and we've applied the Pipeline. How about we move this above the `if requeue` above?